src/app/charts/extra-params-components/threshold.component.ts

```diff
@@ -62,11 +62,11 @@ export class ThresholdComponent implements AfterViewInit, OnInit {
     ngAfterViewInit() {
         // Since valueChanges triggers initially before parent is ready, wait until
         // parent is ready here and trigger it to draw chart with extra parameters.
-        this.thresholdParamSelected.emit({data: {
+        this.thresholdParamSelected.emit({
```

Comment:
Why is the `data` object removed here? You had asked for it to be added in another PR.

Reply:
I think I had misunderstood what the original problem was based on the comments before, combined with forgetting the details of how `EventEmitter`s work. What's here now is the proper way to return data via event emitters, if you do not need the [DOM Event](https://developer.mozilla.org/en-US/docs/Web/API/Event)